### DISCUSSION TITLE
Find/Replace in entire model

### DISCUSSION THREAD

**Discussion by DaniilMaslyuk** [TabularEditor/TabularEditor3#1406] [Ideas]
### Is your feature request related to a problem? Please describe

When I need to replace, say, a variable name in the entire model, and I use Ctrl+H (Find/Replace), I only see two options in the Look in dropdown:

- Current document
- All open documents

There's no Entire model option, even though when I search Ctrl+F, there's an option to look in the entire model.

### Describe the solution you'd like

Make the Look in dropdown in Find and Replace tabs consistent: add Entire model option to the Replace tab

### Goals

Being able to make a model-wide replacement, because sometimes that's exactly what I want, and I don't want to write a C# script for it

**Comment by otykier** [comment]
Hi @DaniilMaslyuk 

Thanks for this suggestion. The current behavior is actually by design - we don't have a notion of performing a replace in the model - only in documents. The reason for that is, that it's not clear which properties such an operation should apply to: Do we only replace in DAX expressions? What about M, or SQL expressions? What about descriptions, names, format strings, display folders, annotations, etc. etc.... so to be on the safe side, and to make sure we don't perform unintended replacements, we limit the replace feature to documents only.

In order to then quickly rename a DAX variable across all DAX in the model, you can use the **Script DAX** option at the model level, then perform the replace operation in that document.

**Comment by DaniilMaslyuk** [comment]
Hi @otykier, thanks for explaining -- you convinced me my desire isn't well thought-out :-) Also thanks for mentioning Script DAX -- I think it'll work for me.
